vispy/visuals/image.py

```diff
@@ -89,12 +89,26 @@
 
 _apply_clim_float = """
     float apply_clim(float data) {
+        if (!(color.r <= 0.0 || 0.0 <= color.r)) {
```

Comment:
Just curious why `(color.r == color.r)` would not work for isnan. I guess the compiler optimizes it away? Anyway, what you have here seems the most suggested solution.

Reply:
Yes. I think that's what I discovered when I originally did this and is why I went with this stackoverflow solution.